geonode_mapstore_client/templates/geonode-mapstore-client/_geonode_config.html

```diff
@@ -132,11 +178,6 @@
                     defaultThumbnailSize: defaultThumbnailSize,
                     upload: { 
                         supportedDatasetFileTypes: supportedDatasetFileTypes
-                        }
-                },
-                geoNodeConfiguration: {
```

Comment:
What happened to the geoNodeConfiguration? The [AppUtils setup](https://github.com/GeoNode/geonode-mapstore-client/blob/4.1.x/geonode_mapstore_client/client/js/utils/AppUtils.js#L318) reads the config which contained the filters and cards configuration which became explicit keys here in this commit (lines 159 and 160). Where is this config read and actually passed? I used this to add custom filters and to create new resource buttons. Am I missing something?

Reply:
@ridoo We created [a plugin for RersoucesGrid in the localConfig.json](https://github.com/GeoNode/geonode-mapstore-client/blob/407984ea446748f429c96747e2466dd7af4061c5/geonode_mapstore_client/client/static/mapstore/configs/localConfig.json#L2643-L2649) so the configuration just moved to the plugin configuration instead to be global. [Here the properties](https://github.com/GeoNode/geonode-mapstore-client/blob/407984ea446748f429c96747e2466dd7af4061c5/geonode_mapstore_client/client/js/plugins/ResourcesGrid.jsx#L165) that could be added in the cfg object including the list of filters and menu items:

- [cfg.filtersFormItems](https://github.com/GeoNode/geonode-mapstore-client/blob/407984ea446748f429c96747e2466dd7af4061c5/geonode_mapstore_client/client/js/plugins/ResourcesGrid.jsx#L222) 
- [cfg,menuItems](https://github.com/GeoNode/geonode-mapstore-client/blob/407984ea446748f429c96747e2466dd7af4061c5/geonode_mapstore_client/client/js/plugins/ResourcesGrid.jsx#L165)

Here you can find also doc on how to [modify the localConfig in a geonode project via js](https://geonode.org/geonode-mapstore-client/4.1.x/tutorial-03-override-local-config.html) or you can use the [MAPSTORE_PLUGINS_CONFIG_PATCH_RULES](https://github.com/GeoNode/geonode/blob/821263674d9271a7f75ebb5027de94d6a16b4730/geonode/settings.py#L1592-L1621) variable in settings.py to change some part of the plugins section of localConfig file configuration:

```
# example not tested
MAPSTORE_PLUGINS_CONFIG_PATCH_RULES = [
    {
        "op": "replace",
        "jsonpath": "$.catalogue..[?(@.name == 'ResourcesGrid')].cfg",
        "value": {
            "containerSelector": "#gn-catalogue",
            "enableGeoNodeCardsMenuItems": true,
            "menuItems": [],
            "filtersFormItems": []
        }
    }
]
```

This change allowed to have different instances of the ResourceGrid with different type of configuration, now the homepage, usually customized, has [its own configuration](https://github.com/GeoNode/geonode-mapstore-client/blob/407984ea446748f429c96747e2466dd7af4061c5/geonode_mapstore_client/templates/index.html#L56-L85) and we added this [doc to show how to create a custom resource page](https://geonode.org/geonode-mapstore-client/4.1.x/tutorial-07-custom-resources-page.html)

Reply:
@allyoucanmap thanks for clarification. I saw the refactoring which allows better flexibility, but missed where the config went. Became the `geonodeConfiguration` in `AppUtils` (passed to the actual apps) an orphaned property then?